docs/reference/files/charmcraft-yaml-file.rst

```diff
@@ -424,6 +432,46 @@ secret URI.
         :start-at: config:
         :end-before: containers:
 
+.. admonition:: Best Practice
+    :class: hint
+
+    Charms should keep the user experience of the operator as simple and obvious
+    as possible. Aim to make the application deployable without providing any
+    further configuration options: ``juju deploy foo`` is preferable to
+    ``juju deploy foo --config required-option=value``.
+
+.. admonition:: Best Practice
+    :class: hint
+
+    Consider which of your application's configuration options you should
+    initially expose. If your application has many config options, it may be
+    prudent to provide access to a select few, and add support for more as the
+    need arises.
+
+.. admonition:: Best Practice
+    :class: hint
+
+    Prefer lowercase alphanumeric names, separated with dashes if required. For
+    charms that have already standardised on underscores, it is not necessary to
+    change them, and it is better to be consistent within a charm then to have
+    some config names be dashed and some be underscored.
+
+.. admonition:: Best Practice
+    :class: hint
+
+    The description a string type. YAML supports two types of formats for that:
```

Comment:
```suggestion
    The description is a string type. YAML supports two types of formats for that:
```

Reply:
I understand that this in the original Best practice doc this was about the config key. However, I feel like this is a comment on the description keys in charmcraft.yaml more generally, so many it's really a best practice at the level of the entire doc. Speaking of which, one more candidate for that top-level note might be https://discourse.charmhub.io/t/yaml-anchors-and-aliases/5683 , which we used to have in the charm SDK docs and I never ported into Charmcraft docs because I wasn't sure about the relevance. @lengau Can you please advise?